buildSrc/src/main/kotlin/org/gradle/plugins/compile/AvailableJavaInstallations.kt

```diff
@@ -26,12 +30,30 @@ class DefaultJavaInstallation(val current: Boolean, private val javaHome: File)
     override fun toString(): String = "${displayName} (${javaHome.absolutePath})"
 }
 
-open class AvailableJavaInstallations(javaHomesForCompilation: List<String>, javaHomeForTest: String?, private val javaInstallationProbe: JavaInstallationProbe) {
-    private val javaInstallations: Map<JavaVersion, DefaultJavaInstallation> = findJavaInstallations(javaHomesForCompilation)
+private
+const val java7HomePropertyName = "java7Home"
+private
+const val testJavaHomePropertyName = "testJavaHome"
+private
+const val oracleJdk8 = "Oracle JDK 8"
+private
+const val oracleJdk7 = "Oracle JDK 7"
+
+open class AvailableJavaInstallations(project: Project, private val javaInstallationProbe: JavaInstallationProbe) {
+
+    private
+    val logger = LoggerFactory.getLogger(AvailableJavaInstallations::class.java)
```

Comment:
This should be static no?

Reply:
I could move it to the companion object, yes.
Given that we only have one instance of `AvailableJavaInstallations` in our build it does not really matter.
Do you feel strongly about this?

Reply:
No.